altered VRControls.js to test for device.deviceName before lowercasing

pindiespace opened the pull request:
test for device.deviceName before lowercasing to avoid error on desktop
browsers.
#7855

pindiespace committed:
altered VRControls.js to test for device.deviceName before lowercasing

test for device.deviceName before lowercasing to avoid error on desktop
browsers.

mrdoob commented on examples/js/controls/VRControls.js:
```diff
@@ -9,22 +9,28 @@ THREE.VRControls = function ( object, onError ) {
 	var vrInputs = [];
 
-	function filterInvalidDevices( devices ) {
+function filterInvalidDevices( devices ) {
```
Oh? Why did you removed the indentation?

pindiespace replied:
my error, no reason.

mrdoob commented on examples/js/controls/VRControls.js:
```diff
@@ -9,22 +9,28 @@ THREE.VRControls = function ( object, onError ) {
+			if(device.deviceName) {
+				return device.deviceName.toLowerCase().indexOf( 'oculus' ) !== - 1;
+			} else {
+				return [];
```
Should't this be ```return false;```?

pindiespace replied:
Later on, we test device.length

devices = filterInvalidDevices( devices );
for ( var i = 0; i < devices.length; i ++ ) {

mrdoob replied:
I think the function passed to Array.filter() is supposed to return a boolean.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

pindiespace replied:
In that case, the function filterInvalidDevices should be changes so it
doesn't assume the returned value has a .length. That would also fix the
broken promise.

On Tue, Dec 29, 2015 at 7:23 AM, Mr.doob <notifications@github.com> wrote:

> In examples/js/controls/VRControls.js
> <https://github.com/mrdoob/three.js/pull/7862#discussion_r48546676>:
>
> >
> >  		// Exclude Cardboard position sensor if Oculus exists.
> >
> >  		var oculusDevices = devices.filter( function ( device ) {
> > -
> > -			return device.deviceName.toLowerCase().indexOf( 'oculus' ) !== - 1;
> > +			if(device.deviceName) {
> > +				return device.deviceName.toLowerCase().indexOf( 'oculus' ) !== - 1;
> > +			} else {
> > +				return [];
>
> I think the function passed to Array.filter() is supposed to return a
> boolean.
>
> https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mrdoob/three.js/pull/7862/files#r48546676>.
>



-- 
============================================================
=================
*Dr. Pete Markiewicz*
*Email:* pindiespace@gmail.com
*Sustainable virtual design blog:*
http://sustainablevirtualdesign.wordpress.com
*Sustainability template:* http://greenboilerplate.com
*Teaching site: *http://plyojump.com <http://www.plyojump.com/>
*Buy my book! *- Millennials and the Popular Culture
*Lifecourse:* http://www.lifecourse.com/store/catalog/lca/mpc.html
* AmazonAuthor Page:*
http://www.amazon.com/Pete-Markiewicz/e/B0069VZ7BG/ref=ntt_athr_dp_pel_pop_2
*On that page, see my DARPA Grand Challenge 2005 entry video, "Team
Robomonster":*
*Team Robomonster *(DARPA 2005 Grand Challenge):
http://robotsthatjump.wordpress.com/category/team-robomonster/
*Indiespace *(1st to sell downloaded music, March 1994):
http://www.indiespace.com. Also see http://archive.org under
"Kspace.com" 1996 for our original site.
============================================================
=================

pindiespace committed:
added field of view (FOV) setter

added setter to change Field of View after object initialized. Helps in
some device orientation changes, and screens in portrait mode.

mrdoob commented:
@borismus @brianpeiris is the `filterInvalidDevices` method still needed? I thought it was a temporary thing?

pindiespace commented:
Possibly...why is deviceName needed in any case? (instead of the id)

On Mon, Jan 11, 2016 at 5:40 AM, Mr.doob <notifications@github.com> wrote:

> @borismus <https://github.com/borismus> @brianpeiris
> <https://github.com/brianpeiris> is the filterInvalidDevices method still
> needed? I thought it was a temporary thing?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mrdoob/three.js/pull/7862#issuecomment-170551487>.
>



-- 
============================================================
=================
*Dr. Pete Markiewicz*
*Email:* pindiespace@gmail.com
*Sustainable virtual design blog:*
http://sustainablevirtualdesign.wordpress.com
*Sustainability template:* http://greenboilerplate.com
*Teaching site: *http://plyojump.com <http://www.plyojump.com/>
*Buy my book! *- Millennials and the Popular Culture
*Lifecourse:* http://www.lifecourse.com/store/catalog/lca/mpc.html
* AmazonAuthor Page:*
http://www.amazon.com/Pete-Markiewicz/e/B0069VZ7BG/ref=ntt_athr_dp_pel_pop_2
*On that page, see my DARPA Grand Challenge 2005 entry video, "Team
Robomonster":*
*Team Robomonster *(DARPA 2005 Grand Challenge):
http://robotsthatjump.wordpress.com/category/team-robomonster/
*Indiespace *(1st to sell downloaded music, March 1994):
http://www.indiespace.com. Also see http://archive.org under
"Kspace.com" 1996 for our original site.
============================================================
=================

brianpeiris commented:
It looks like `filterInvalidDevices` is not needed anymore since Firefox Nightly no longer returns a phantom Google Cardboard VR device on desktops. They must have fixed that bug. See #6340 and #6380 for reference. 

I think it's safe to remove it unless someone at MozVR says otherwise. @dmarcos?

pindiespace commented:
I've had problems outside of Nighly. I've been using Edge's Windows Phone
emulation, which works well, and doesn't require dowloading a dev browser
(so you can use it on any computer, i.e. my classrooms). I get the error
there. Removing filterInvalidDevices would fix.

On Mon, Jan 11, 2016 at 12:12 PM, Brian Peiris <notifications@github.com>
wrote:

> It looks like filterInvalidDevices is not needed anymore since Firefox
> Nightly no longer returns a phantom Google Cardboard VR device on desktops.
> They must have fixed that bug. See #6340
> <https://github.com/mrdoob/three.js/pull/6340> and #6380
> <https://github.com/mrdoob/three.js/pull/6380> for reference.
>
> I think it's safe to remove it unless someone at MozVR says otherwise.
> @dmarcos <https://github.com/dmarcos>?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mrdoob/three.js/pull/7862#issuecomment-170675182>.
>



-- 
============================================================
=================
*Dr. Pete Markiewicz*
*Email:* pindiespace@gmail.com
*Sustainable virtual design blog:*
http://sustainablevirtualdesign.wordpress.com
*Sustainability template:* http://greenboilerplate.com
*Teaching site: *http://plyojump.com <http://www.plyojump.com/>
*Buy my book! *- Millennials and the Popular Culture
*Lifecourse:* http://www.lifecourse.com/store/catalog/lca/mpc.html
* AmazonAuthor Page:*
http://www.amazon.com/Pete-Markiewicz/e/B0069VZ7BG/ref=ntt_athr_dp_pel_pop_2
*On that page, see my DARPA Grand Challenge 2005 entry video, "Team
Robomonster":*
*Team Robomonster *(DARPA 2005 Grand Challenge):
http://robotsthatjump.wordpress.com/category/team-robomonster/
*Indiespace *(1st to sell downloaded music, March 1994):
http://www.indiespace.com. Also see http://archive.org under
"Kspace.com" 1996 for our original site.
============================================================
=================

brianpeiris commented:
Yeah, I'd say go ahead and remove it, seems like a sensible thing to do. If Firefox regresses, it's their bug.

mrdoob commented:
Done. Thanks!
